src/ConnectionAutomaton.cpp

```diff
@@ -293,6 +310,17 @@ void ConnectionAutomaton::onConnectionFrame(
       }
       return;
     }
+    case FrameType::ERROR: {
+      Frame_ERROR frame;
+      if (frame.deserializeFrom(std::move(payload))) {
+        if (frame.errorCode_ == ErrorCode::CONNECTION_ERROR && !isServer_ && isResumable_ && resumeStatusCallback_) {
+          resumeStatusCallback_->onConnectionError(std::runtime_error(frame.payload_.moveDataToString()));
+          resumeStatusCallback_.reset();
```

Comment:
how about the case when ReactiveSocket instance is released while the resumeStatusCallback is in flight? Maybe the disconnect method or dtor should fail the resumeStatusCallback_